pkg/anchor/writer/writer.go

```diff
@@ -214,6 +224,48 @@ func (c *Writer) handle(vc *verifiable.Credential) {
 	c.txnCh <- []string{cid}
 }
 
+// getWitnesses returns the list of anchor origins for all dids in the Sidetree batch.
+// Create and recover operations contain anchor origin in operation references.
+// For update and deactivate operations we have to 'resolve' did in order to figure out anchor origin.
+func (c *Writer) getWitnesses(refs []*operation.Reference) ([]string, error) {
+	var witnesses []string
+
+	uniqueWitnesses := make(map[string]bool)
+
+	for _, ref := range refs {
+		var anchorOriginObj interface{}
+
+		switch ref.Type {
+		case operation.TypeCreate, operation.TypeRecover:
+			anchorOriginObj = ref.AnchorOrigin
+
+		case operation.TypeUpdate, operation.TypeDeactivate:
+			result, err := c.OpProcessor.Resolve(ref.UniqueSuffix)
+			if err != nil {
+				return nil, err
+			}
+
+			anchorOriginObj = result.AnchorOrigin
+		default:
+			return nil, fmt.Errorf("operation type '%s' not supported for assembling witness list", ref.Type)
+		}
+
+		// TODO: string or array of strings?
```

Comment:
Follow up issue #?

Reply:
This is more question for Troy, not an issue. I will talk to him about it and fix it in one of my future PRs.